Wikipedia/Code/WelcomeAnimationExtensions.swift

```diff
@@ -0,0 +1,753 @@
+        shape.lineDashPattern = [toView.frame.size.width * 0.029, toView.frame.size.width * 0.047]
+        shape.transform = circle.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: circle.position, initialOpacity: circle.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addCircle(circle: WelcomeCircle, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_circlePathWithRadius(toView.frame.size.width * circle.radius).CGPath
+        shape.fillColor = UIColor.blackColor().CGColor
+        shape.strokeColor = UIColor.clearColor().CGColor
+        shape.transform = circle.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: circle.position, initialOpacity: circle.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addPlus(plus: WelcomePlus, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        shape.path = UIBezierPath.wmf_plusPathWithWidth(toView.frame.size.width * 0.04).CGPath
+        shape.strokeColor = UIColor.blackColor().CGColor
+        shape.transform = plus.animation.initialTransform
+        shape.wmf_applyCommonSettingsForSize(toView.frame.size, position: plus.position, initialOpacity: plus.animation.initialOpacity)
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+    class func wmf_addBar(bar: WelcomeBar, toView: UIView) -> CAShapeLayer {
+        let shape = CAShapeLayer()
+        let rect = CGRectMake(
+            0.0,
+            0.0,
+            toView.frame.size.width * bar.frame.size.width,
+            toView.frame.size.height * bar.frame.size.height
+        )
+        shape.path = UIBezierPath(roundedRect: rect, cornerRadius: bar.cornerRadius).CGPath
+        shape.fillColor = bar.color.CGColor
+        shape.transform = bar.animation.initialTransform
+        shape.position = CGPointMake(
+            (toView.frame.size.width * bar.frame.origin.x),
+            (toView.frame.size.height * bar.frame.origin.y)
+        )
+        toView.layer.addSublayer(shape)
+        return shape;
+    }
+}
+
+extension CALayer{
+    func wmf_addWelcomeAnimation(animation: WelcomeAnimation) {
+        self.transform = animation.initialTransform;
+        self.addAnimation(CABasicAnimation.wmf_animationToXF(animation.finalTransform, delay: animation.delay, duration: animation.duration), forKey: nil)
+        
+        self.opacity = Float(animation.initialOpacity);
+        self.addAnimation(CABasicAnimation.wmf_animationToOpacity(animation.finalOpacity, delay: animation.delay, duration: animation.duration), forKey: nil)
+    }
+}
+
+extension CABasicAnimation{
+    class func wmf_animationToXF(transform: CATransform3D, delay: Double, duration: Double) -> CABasicAnimation {
```

Comment:
Not sure what XF is… remember spell it out.

Also like above, you are not adding an animation… you are building one and returning it. (a constructor)